app/controllers/api/v1/cities_controller.rb

```diff
@@ -2,4 +2,17 @@ class Api::V1::CitiesController < ApplicationController
   def index
     render json: City.all
   end
+
+  # TODO must be signed in
+  def create
+    city              = City.new(params.require(:city).permit(:name))
+    city.timezone     = "Pacific Time (US & Canada)" if !city.timezone
+    city.city_code    = CityCodeGenerator.generate if !city.city_code
```

Comment:
if `city.city_code.nil?` would be more idiomatic. As would it being `city.code`. Yo dawg…